src/services/userService.js

```diff
@@ -23,11 +29,56 @@ module.exports = {
       realLifeName,
       reputationScore,
       createdAt,
-      votingPower
+      votingPower,
+      steemBalance,
+      sbdBalance,
+      userVests,
     }
   },
 
-  getUserInfo: async function(name) {
+  getUserProfileInfo: async function(name) {
+    return this.getUserAccount(name)
+      .then(userInOurDb => {
+        return this.getDynamicVestInfo(userInOurDb);
+      })
+      .then(userInOurDb => {
+        return this.getCoinMarketCapPrices(userInOurDb);
+      })
+      .catch(err => {
+        new Error(err);
+      })
+  },
+
+  getCoinMarketCapPrices: async function(userRecord) {
+    const url = "https://api.coinmarketcap.com/v1/ticker/steem";
+
+    return await fetch(url)
+      .then(function(data) {
+          const steemPriceInUsd = parseFloat(data.price_usd);
+          const steemTokensInAccount = userRecord.steemBalance + userRecord.steemPower
+          const estimatedAccountValue = steemTokensInAccount * steemPriceInUsd + userRecord.sbdBalance;
+          return userRecord.update({ estimatedAccountValue });
+      })
+      .catch(err => {
+        new Error(err);
+      })
+  }
+
+  getDynamicVestInfo: async function(userRecord) {
+    const calculateSteemPower = (vestInfo) => {
+      const vestProperties = vestInfo[0];
+      const totalSteem = parseFloat(vestProperties.total_vesting_fund_steem);
+      const totalVests = parseFloat(vestProperties.total_vesting_shares);
+      const userVests = userRecord.userVests;
+      const steemPower = totalSteem * (userVests / totalVests);
+      return userRecord.update({ steemPower });
+    }
```

Comment:
Let's move the `calculateSteemPower` helper function outside of `getDynamicVestInfo`. It's going to re-create the `calculateSteemPower` function every time we call `getDynamicVestInfo`.

Reply:
done